src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -499,32 +499,48 @@ internal static void CreateSymbolicLink(string path, string pathToTarget, bool i
                 }
 
                 Span<byte> bufferSpan = new(buffer);
-                success = MemoryMarshal.TryRead(bufferSpan, out Interop.Kernel32.REPARSE_DATA_BUFFER rdb);
+                success = MemoryMarshal.TryRead(bufferSpan, out Interop.Kernel32.SymbolicLinkReparseBuffer rbSymlink);
                 Debug.Assert(success);
 
-                // Only symbolic links are supported at the moment.
-                if ((rdb.ReparseTag & Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_SYMLINK) == 0)
-                {
-                    return null;
-                }
-
-                // We use PrintName instead of SubstitutneName given that we don't want to return a NT path when the link wasn't created with such NT path.
+                // We use PrintName(Offset|Length) instead of SubstituteName(Offset|Length) given that we don't want to return
+                // an NT path when the link wasn't created with such NT path.
                 // Unlike SubstituteName and GetFinalPathNameByHandle(), PrintName doesn't start with a prefix.
                 // Another nuance is that SubstituteName does not contain redundant path segments while PrintName does.
-                // PrintName can ONLY return a NT path if the link was created explicitly targeting a file/folder in such way. e.g: mklink /D linkName \??\C:\path\to\target.
-                int printNameNameOffset = sizeof(Interop.Kernel32.REPARSE_DATA_BUFFER) + rdb.ReparseBufferSymbolicLink.PrintNameOffset;
-                int printNameNameLength = rdb.ReparseBufferSymbolicLink.PrintNameLength;
+                // PrintName can ONLY return a NT path if the link was created explicitly targeting a file/folder in such way.
+                //   e.g: mklink /D linkName \??\C:\path\to\target.
+
+                if (rbSymlink.ReparseTag == Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_SYMLINK)
+                {
+                    int printNameOffset = sizeof(Interop.Kernel32.SymbolicLinkReparseBuffer) + rbSymlink.PrintNameOffset;
+                    int printNameLength = rbSymlink.PrintNameLength;
 
-                Span<char> targetPath = MemoryMarshal.Cast<byte, char>(bufferSpan.Slice(printNameNameOffset, printNameNameLength));
-                Debug.Assert((rdb.ReparseBufferSymbolicLink.Flags & Interop.Kernel32.SYMLINK_FLAG_RELATIVE) == 0 || !PathInternal.IsExtended(targetPath));
+                    Span<char> targetPathSymlink = MemoryMarshal.Cast<byte, char>(bufferSpan.Slice(printNameOffset, printNameLength));
```

Comment:
```suggestion
                    Span<char> targetPath = MemoryMarshal.Cast<byte, char>(bufferSpan.Slice(printNameOffset, printNameLength));
```
nit: There's no value in explain which reparse point was holding the target, the same goes for targetPathMountPoint.